Math/NumberTheory/Moduli/PrimitiveRoot.hs

```diff
@@ -97,6 +101,18 @@ cyclicGroupToModulo = fromFactors . \case
   CGOddPrimePower p k       -> [(unPrime p, k)]
   CGDoubleOddPrimePower p k -> [(2, 1), (unPrime p, k)]
 
+-- | 'PrimitiveRoot n' is a type which is only inhabited by primitive roots of
+-- n.
+newtype PrimitiveRoot n = PrimitiveRoot (Mod n)
+  deriving (Eq, Ord, Show)
```

Comment:
Let us define `newtype PrimitiveRoot n = PrimitiveRoot { unprimitiveRoot :: Mod n }` and export the type and the accessor, but not the constructor.